packages/flutter/lib/src/material/time.dart

```diff
@@ -185,7 +185,7 @@ HourFormat hourFormat({ @required TimeOfDayFormat of }) {
   switch (of) {
     case TimeOfDayFormat.h_colon_mm_space_a:
     case TimeOfDayFormat.a_space_h_colon_mm:
-      return HourFormat.h;
+      return window.alwaysUse24HourFormat ? HourFormat.HH : HourFormat.h;
```

Comment:
this code doesn't register to be told when the flag changes, which seems bad. I would recommend putting this flag in the MediaQueryData object and having the caller read that then pass in the flag to this method.